drv/lpc55-update-server/src/main.rs

```diff
@@ -113,17 +113,25 @@ impl idl::InOrderUpdateImpl for ServerImpl {
 
         let img = self.image.unwrap_lite();
 
-        match tz_table!().write_to_flash(
-            img,
-            block_num as u32,
-            flash_page.as_mut_ptr(),
-        ) {
-            HypoStatus::Success => Ok(()),
-            HypoStatus::OutOfBounds => Err(UpdateError::OutOfBounds.into()),
-            HypoStatus::RunningImage => Err(UpdateError::RunningImage.into()),
-            // Should probably encode the LPC55 flash status into the update
-            // error for good measure but that takes effort...
-            HypoStatus::FlashError(_) => Err(UpdateError::FlashError.into()),
+        unsafe {
+            // The write_to_flash API takes raw pointers due to TrustZone
+            // ABI requirements which makes this function unsafe.
+            match tz_table!().write_to_flash(
```

Comment:
I think we could shrink the `unsafe` block a bit like this:

```rust
let result = unsafe {
    tz_table!().write_to_flash(...)
};
match result {
    ...
}
```

since it's only needed to call the function, not deal with the result.

Reply:
good suggestion